action/protocol/staking/endorsement.go

```diff
@@ -0,0 +1,70 @@
+package staking
+
+import (
+	"math"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/iotexproject/iotex-core/action/protocol/staking/stakingpb"
+)
+
+// EndorsementStatus
+const (
+	NotEndorsed = EndorsementStatus(iota)
+	UnEndorsing
+	Endorsed
+)
+
+const (
+	endorsementNotExpireHeight = math.MaxUint64
+)
+
+type (
+	// EndorsementStatus is a uint8 that represents the status of the endorsement
+	EndorsementStatus uint8
+
+	// Endorsement is a struct that contains the expire height of the Endorsement
+	Endorsement struct {
+		ExpireHeight uint64
```

Comment:
Is there only one field?

Reply:
yes, `bucketIndex` is the key for `Endorsement`, and endorser and candidate should be the owner and candidate of the bucket